app/services/integrations/aggregator/base_service.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require 'lago_http_client'
+
+module Integrations
+  module Aggregator
+    class BaseService < BaseService
+      BASE_URL = 'https://api.nango.dev/'
+
+      def initialize(integration:)
+        @integration = integration
+
+        super
+      end
+
+      def action_path
+        raise NotImplementedError
+      end
+
+      private
+
+      attr_reader :integration
+
+      def http_client
+        LagoHttpClient::Client.new(endpoint_url)
+      end
+
+      def endpoint_url
+        "#{BASE_URL}#{action_path}"
+      end
+
+      def generate_headers
```

Comment:
I'd call it just headers, but it's up to you.

```suggestion
      def headers
```